### DISCUSSION TITLE
Add option for embedded inline link previews

### DISCUSSION THREAD

**Discussion by ghost** [Chatterino/chatterino2#4695] [Ideas]
Hovering links for a preview can be difficult when doing something else at the same time (streaming/gaming) and unnecessary on 1:1 conversations, embedded previews would help with that. 
Example: 
![LXJoa](https://github.com/Chatterino/chatterino2/assets/51298186/57a8ab66-9b34-4729-a2fe-c316d3e06462)
Edit: to be clear I am talking about clip previews mainly.

**Comment by leon-richardt** [comment]
This idea has been brought up before (at least "internally") and I remember there being support for it.

Before this can be implemented in a reasonable manner though, the link resolver in [Chatterino/api](https://github.com/Chatterino/api) would have to return link information in some easily-parsable format (e.g. JSON) instead of returning raw HTML which the client just renders immediately. Using JSON, users could decide for themselves whether they prefer the legacy hover-only or the supposed new embedded previews.

EDIT: I have created https://github.com/Chatterino/api/issues/498 to track the idea and gather implementation notes.
